workload/pkg/info.go

```diff
@@ -224,6 +218,146 @@ func CreateResourceProviderFromAPI(ctx context.Context, kube kubernetes.Interfac
 	}
 
+	controllerMap := map[string]*ControllerResult{}
+	for idx := range interfaces {
+		controllerMap[interfaces[idx].UID] = &interfaces[idx]
+	}
+	topController := map[string]bool{}
+	children := map[string][]string{}
+	for _, pointer := range controllerMap {
+		lastParent := pointer
+		for lastParent.ParentUID != "" {
+			newParent := controllerMap[lastParent.ParentUID]
+			if newParent == nil {
+				break
+			}
+			lastParent = newParent
+		}
+		topController[lastParent.UID] = true
+		if pointer.ParentUID != "" {
+			children[lastParent.UID] = append(children[lastParent.UID], pointer.UID)
+		}
+	}
+
+	type jobMetadata struct {
+		startTime time.Time
+		endTime   time.Time
+	}
+	cronChildren := map[string][]jobMetadata{}
+	// Jobs
+	jobs, err := kube.BatchV1().Jobs("").List(ctx, listOpts)
+	if err != nil {
+		logrus.Errorf("Error fetching Jobs %v", err)
+		return nil, err
+	}
+
+	for _, item := range jobs.Items {
+		var containers []ContainerResult
+		if item.Status.Active+item.Status.Failed > 0 {
+			continue
+		}
+		ownerUID := getOwnerUID(item.GetObjectMeta().GetOwnerReferences())
+		if ownerUID != "" {
+			ownerController := controllerMap[ownerUID]
+			if ownerController != nil {
+				if item.Status.StartTime != nil && item.Status.CompletionTime != nil {
+					cronChildren[ownerUID] = append(cronChildren[ownerUID], jobMetadata{
+						startTime: item.Status.StartTime.Time,
+						endTime:   item.Status.CompletionTime.Time,
+					})
+				}
+				continue
+			}
+		}
+		for _, container := range item.Spec.Template.Spec.Containers {
+			containers = append(containers, formatContainer(container, corev1.ContainerStatus{}, item.Spec.Template.GetCreationTimestamp()))
+		}
+		job := formatControllers("Job", item.Name, item.Namespace, string(item.UID), item.GetObjectMeta().GetOwnerReferences(), containers, item.Annotations, item.Labels)
+		interfaces = append(interfaces, job)
+	}
+
+	// Pods
+	pods, err := kube.CoreV1().Pods("").List(ctx, listOpts)
+	if err != nil {
+		logrus.Errorf("Error fetching Pods %v", err)
+		return nil, err
+	}
+
+	for _, item := range pods.Items {
+		var containers []ContainerResult
+		if item.Status.Phase != corev1.PodRunning && item.Status.Phase != corev1.PodPending {
+			continue
+		}
+		for _, container := range item.Spec.Containers {
+			for _, status := range item.Status.ContainerStatuses {
+				if status.Name == container.Name {
+					containers = append(containers, formatContainer(container, status, item.GetObjectMeta().GetCreationTimestamp()))
+					break
+				}
+			}
+		}
+		ownerReferences := item.GetObjectMeta().GetOwnerReferences()
+		ownerUID := getOwnerUID(ownerReferences)
+
+		if ownerUID != "" {
+			ownerController := controllerMap[ownerUID]
+			if ownerController != nil {
+				controllerMap[ownerUID].PodCount++
+				ownerContainers := controllerMap[ownerUID].Containers
+				if len(ownerContainers) == 0 || (len(containers) > 0 && containers[0].CreationTime.After(ownerContainers[0].CreationTime)) {
```

Comment:
not sure I understand why we swap out ownerContainers for podContainers - when would they differ?

Reply:
The container status (including Image ID) isn't there at the Deployment level, so this tries to get the latest Pod and use that for container status.